### PR TITLE
chore: merge release 7.29.1 into main

### PR DESCRIPTION
**Pull request by Daniel-Cross**
## **Description**

Branch of current main with release/7.29.1 merged in and conflicts resolved.

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -190,5 +190,5 @@ module.exports = {
     radix: 0,
   },
 
-  ignorePatterns: ['wdio.conf.js', 'app/util/termsOfUse/termsOfUseContent.ts'],
```
Comment (sethkfman) [inline .eslintrc.js:193]: We should keep this a ts file.

File: .iyarc [hunk 2]
```diff
@@ -1,3 +1,4 @@
 # ReDoS vulnerability, no impact to this application, and fix not backported yet to the versions we use
 
 GHSA-c2qf-rxjj-qqgw
```
Comment (sethkfman) [inline .iyarc:3]: Need to remove: `GHSA-8hc4-vh64-cxmj`

File: .eslintrc.js (hunk 1 above)
Comment (Daniel-Cross) [reply .eslintrc.js:193]: Good spot, I couldn't see a difference in these two lines

File: patches/@metamask+swaps-controller+9.0.1.patch [hunk 3]
```diff
@@ -0,0 +1,15 @@
+diff --git a/node_modules/@metamask/swaps-controller/dist/constants.js b/node_modules/@metamask/swaps-controller/dist/constants.js
+index 4c98522..523f25a 100644
+--- a/node_modules/@metamask/swaps-controller/dist/constants.js
++++ b/node_modules/@metamask/swaps-controller/dist/constants.js
```
Comment (sethkfman) [inline patches/@metamask+swaps-controller+9.0.1.patch:4]: We should remove this patch.

**Review by sethkfman — APPROVED** [review]
LGTM

**Pull request merged**